### PR TITLE
fix wrong importance sampling ratio logging.

### PR DESCRIPTION
**Pull request by lydiadli**
**What does this PR do? Please describe:**
fix wrong importance sampling ratio metric - current calculation include positions supposed to be masked out including padding and prompt tokens. metric is abnormal even for quite on-policy setup.

Fixes #{issue number}

**Does your PR introduce any breaking changes? If yes, please list them:**
List of all backwards-incompatible changes.

**Check list:**
- [ ] Was the content of this PR **discussed and approved** via a GitHub issue? (no need for typos or documentation improvements)
- [X] Did you read the [contributor guideline](https://github.com/facebookresearch/fairseq2/blob/main/CONTRIBUTING.md)?
- [ ] Did you make sure that your **PR does only one thing** instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [X] Did you write any **new necessary tests**?
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you **update the [CHANGELOG](https://github.com/facebookresearch/fairseq2/blob/main/CHANGELOG.md)**? (no need for typos, documentation, or minor internal changes)

### CODE REVIEW

File: src/fairseq2/recipes/lm/_online_finetune/_common.py [hunk 1]
```diff
@@ -746,3 +746,10 @@ def get_parameter_converter(model_config):
         raise RuntimeError(f"{model_config} not supported in online recipe")
 
     return _convert_parameter
+
+
+def masked_mean(x, mask, eps=1e-8):
```
Comment (uralik) [inline src/fairseq2/recipes/lm/_online_finetune/_common.py:751]: this eps value wont work with fp16 and will lead to funny numbers
what you can do is to use x dtype smallest value that pytorch can generate for you

Comment (lydiadli) [reply src/fairseq2/recipes/lm/_online_finetune/_common.py:751]: thanks for the thoughtfulness! this case should be ok actually i think. because whenever denominator is indeed clamped, mask only has zeros. so the masked_x (numerator) is zero as well.  it would be torch.tensor([0], dtype=torch.float16) / 1e-8 or similar for other datetypes :)
